src/js/components/Layout/menu/Menu.jsx

```diff
@@ -31,27 +31,23 @@ const Menu = ({ menuConfig, location }) => {
   }, {});
 
   const checkActiveSection = (menuUrls, path) => {
-    // Concat currentPath, because react-router recognizes params (?exampleParam)
-    // and it moves this part from pathname and moves it to search
-    // e.g.
-    // /openboxes/stockMovement/createInbound?directon=INBOUND
-    // =>
-    // pathname: /openboxes/stockMovement/createInbound/ search: ?direction=INBOUND
-    const currentPath = path.pathname.concat('', path.search);
-    // slice currentPath to max length of 4,
-    // so we look only at e.g. /openboxes/invoice/create/,
-    // so we don't need to check additional /invoicing/create/:id/:param1/:param2, etc.
-    const splittedCurrentPath = currentPath.split('/').slice(0, 4);
-    const matchedPath = Object.keys(menuUrls).find((section) => {
-      const splittedSectionUrls = menuUrls[section].map(url => url.split('/').slice(0, 4));
-      // Check if any of section's spllited url matches currentPath
-      return splittedSectionUrls.some((url) => {
-        if (splittedCurrentPath.length === url.length) {
-          return !!splittedCurrentPath.every((el, idx) => el === url[idx]);
-        }
-        return false;
+    const { pathname, search } = path;
+    const matchedPath = Object.keys(menuUrls)
+      .find((section) => {
+        // find matching URL from sections
+        const foundURL = menuUrls[section].find((url) => {
+          const [sectionPath, sectionSearch] = url.split('?');
+          if (sectionPath !== pathname) return false;
+          // if found matching pathname
+          // then check if all parameters of section path match with current path parameters
+          if (sectionSearch) {
+            const searchParams = sectionSearch.split('&');
+            return search && searchParams.every(param => search.includes(param));
+          }
+          return true;
+        });
+        return !!foundURL;
```

Comment:
I needed to refactor this method because it match URLs with path that had search parameters in different order.
The problem that I noticed was with INBOUND/OUTBOUND stock movement

In example in config for `menuUrls` we have an object where key is the _name_  of the section and _value_ is the array of urls that the section contains.
For Inbound section we have a url for inbound list of `/openboxes/stockMovements/lis?direction=INBOUND`.
In the old method it would select it as active it the current URL was exactly the same as specified in the array, so for example such a URL `/openboxes/stockMovements/lis?status=PENDING&direction=INBOUND`. would not be matched.

In my implementation of this _active section checked_ I check if section has search params we make sure that current URL has all required params and if it does we return it as matched URL.